File: python/stack/evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for i in tokens:
            if i.isnumeric():
                stack.append(int(i))
            elif len(i) > 1 and i[0] == "-":
                stack.append(-(int(i[1:])))
            else:
                first = stack.pop()
                second = stack.pop()
                if i == "+":
                    stack.append(int(first + second))
                elif i == "-":
                    stack.append(int(second - first))
                elif i == "*":
                    stack.append(int(first * second))
                else:
                    stack.append(int(second / first))

        return stack[-1]

File: c++/stack/evaluate-reverse-polish-notation.cc
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        std::vector<int> stack;
        std::vector<std::string> operators = {"+", "-", "*", "/"};

        for (std::string token : tokens) {
            if (std::find(operators.begin(), operators.end(), token) == operators.end()) {
                stack.push_back(std::stoi(token));
            } else {
                int input1 = stack.back();
                stack.pop_back();
                int input2 = stack.back();
                stack.pop_back();

                int answer;
                if (token == "+") {
                    answer = input2 + input1;
                } else if (token == "-") {
                    answer = input2 - input1;
                } else if (token == "*") {
                    answer = input2 * input1;
                } else {
                    answer = input2 / input1;
                }
                stack.push_back(answer);
            }
        }
        return stack.back();
    }
};
